cmd/util/ledger/migrations/storage_v4.go

```diff
@@ -66,36 +70,71 @@ func storageFormatV4MigrationWorker(jobs <-chan ledger.Payload, results chan<- s
 		if err != nil {
 			result.error = err
 		} else {
+			if err := checkStorageFormatV4(migratedPayload); err != nil {
+				panic(fmt.Errorf("%w: key = %s", err, payload.Key.String()))
+			}
 			result.Payload = migratedPayload
 		}
 		results <- result
 	}
 }
 
+var decMode = func() cbor.DecMode {
+	decMode, err := cbor.DecOptions{
+		IntDec:           cbor.IntDecConvertNone,
+		MaxArrayElements: math.MaxInt32,
+		MaxMapPairs:      math.MaxInt32,
+		MaxNestedLevels:  256,
```

Comment:
Are these the same params as Cadence storage?

Reply:
No, this mode uses the maximums, as we only use it to determine if some data is CBOR. The Cadence decoder then still uses it's own decoding mode to decode.